app/models/default_pet_task.rb

```diff
@@ -24,4 +25,6 @@ class DefaultPetTask < ApplicationRecord
 
   validates :name, presence: true
   validates_numericality_of :due_in_days, only_integer: true, greater_than_or_equal_to: 0, allow_nil: true
+
+  enum species: {"All" => nil}.merge(Pet.species)
```

Comment:
```suggestion
    enum :species, ["All"].push(Pet.species.keys).flatten, default: 0
```
I think using`"All" => 0` as mention in the issue would more make sense. Using nil seems to go against the implicit value's and I'm not sure what we gain from that.

Reply:
The reason for using nil here is to maintain the consistency between the "species" enum in Pet model where Dogs have 0 and Cats have 1. Also I did not opt for modifying the enum "species" in Pet model and went with a new one in DefaultPetTask is because adding All as part of pet enum would not be consistent when viewed as a potential value in a single pet instance. So I went with DefaultPetTask. So keeping "All" as nil felt essential in my thought process. Love to hear your take.

Reply:
I see where you are coming from. I didn't think about the value matching on `Pets` and `DefaultPetTasks`. Although the value should rarely if ever be called, would this cause any confusion for future development? Maybe @kasugaijin or @mononoken might have some thoughts to add.

Current
```
Pet.species: {"Dog"=>0, "Cat"=>1} #implicit
DefaultPetTask.species: {"All"=>nil, "Dog"=>0, "Cat"=>1} #explicit
```

Suggestion (original issue criteria)
```
Pet.species: {"Dog"=>0, "Cat"=>1} #implicit
DefaultPetTask.species: {"All"=>0, "Dog"=>1, "Cat"=>2} #implicit
```

Third option
```
Pet.species: {"Dog"=>1, "Cat"=>2} #Explicit to have the same values as DefaultPetTask without relying on a nil value
DefaultPetTask.species: {"All"=>0, "Dog"=>1, "Cat"=>2} #implicit
```
Not part of the issue, but one thought came to mind was the potential need in the future to have a  `none` option. Maybe an admin does not want to delete a `DefaultPetTask` but they do not want it being added to a category temporarily. I think using the nil option in that scenario would make more sense.

Reply:
Ah, this is a good question and harps back on a discussion we had a while back about abstracting the concept of pet species into its own entity to prevent issues like this. We decided on keeping the species on Pet as the source of truth, but this PR identifies the issue with doing that...it causes coupling of models where it ought not to exist. I think the easiest way to fix this for now is to match the enums between the two models, per Justin's third option. The fact we have to change another model to facilitate this underlines the issue...but it should be a quick fix (maybe check the seed files too to update them if they use the integers to set species). Do you agree?